tests/core/io/test_resource.h

```diff
@@ -123,10 +123,14 @@ TEST_CASE("[Resource] Breaking circular references on save") {
 
 	const String save_path_binary = OS::get_singleton()->get_cache_path().path_join("resource.res");
 	const String save_path_text = OS::get_singleton()->get_cache_path().path_join("resource.tres");
+	ERR_PRINT_OFF
 	ResourceSaver::save(resource_a, save_path_binary);
 	ResourceSaver::save(resource_a, save_path_text);
+	ERR_PRINT_ON
```

Comment:
I'm not sure it's correct to silence those. The error message says "bug?", and that means either the message is wrong (this is a normal, expected failure from misusing the API) or there is a bug to be fixed.

Reply:
Then I'll just fix the `has_meta` and leave that for further investigation

Reply:
Tried investigating the exact details in the code when I was working on this yesterday, and wasn't sure exactly where to go with it

Now only fixing the `get_meta` part

The oddity is that the binary case stores the resource reference regardless, as it has parsed it all before storing, but it then gets a warning when reading, can silence that one but I think it might be considered an error that it stores cyclic references

The message in the text case should probably be fixed as well, as it might be missing some details.

Both cases do have seemingly cyclic error detection with dedicated messages but doesn't show up for these